src/vmm/src/devices/virtio/vsock/csm/connection.rs

```diff
@@ -742,6 +750,16 @@ mod tests {
         }
     }
 
+    impl ReadVolatile for TestStream {
+        fn read_volatile<B: BitmapSlice>(
+            &mut self,
+            buf: &mut VolatileSlice<B>,
+        ) -> std::result::Result<usize, utils::vm_memory::VolatileMemoryError> {
+            // Test code, the additional copy incurred by read_from is fine
+            buf.read_from(0, self, buf.len())
```

Comment:
sure, but wouldn't it be better to test even more our `ReadVolatile` implementation (or am I missing something?)

Reply:
In this particular case we cannot use it, as we _have_ to go through the `Read` implementation, which "mocks" out having an actual unixstream and instead just copied to/from memory.